docs/adr/0004-jwt-token-generation.md

```diff
@@ -0,0 +1,49 @@
+# 0004 - JWT Token Generation
+
+Date: April 12, 2021
+
+## Context
+
+The application's REST API uses JWT tokens for authentication.
+
+An important consideration is how these JWT tokens are generated and distributed
+to the servers (ArchiveSpace, Fedora 2, etc.) that need to communicate with
+this application. Also important is the need to identify which applications
+may be using a token, in case it becomes necessary to replace them.
+
+## Decision
+
+Create a Rake task for generating a JWT token, where the generated token
+is recorded in a "token log" table within the application. The "token log"
+table should contain:
+
+* the generated token
```

Comment:
@dsteelma-umd  I thought we were not storing the actual token itself. But, I guess there is no problem with storing that. I just wanted to make sure.

Reply:
@mohideen - From the "Architecture" section in the design document (https://docs.google.com/document/d/1mxIfK-EaXc5zvlpYwTMQFk5HSH-ifYtHaeIeKIOZmYo):

----
* REST
    * Json Web Token (JWT): https://jwt.io/
    * Uses a stored secret to generate and validate the tokens
    * Generate via rake task, store log in db table w/ description (not available through UI):
            * Table fields: token, description, date created
----

I don't see any additional security issues with storing the tokens (in that if an unauthorized person can get to the table containing the tokens we're already in trouble), and I think we might want to in case we someday want to link records as having been generated by a particular token.